### ISSUE TITLE
Close callback being called multiple times when location changes

### ISSUE THREAD

**Issue by DougKeller** [dwmkerr/angular-modal-service#258]
**TL;DR:** If the location changes while waiting on an async operation before closing a modal, the `cleanUpClose` function is invoked multiple times, resulting in an exception:
>TypeError: Cannot read property 'resolve' of null ([line 206 of angular-modal-service.js](https://github.com/dwmkerr/angular-modal-service/blob/master/src/angular-modal-service.js#L206))

**Steps to Reproduce**
1. perform an async operation within a modal's controller, closing the modal once it's complete
```javascript
// ModalController
$http.post('something').then(() => {
  close();
});
```
2. While the request is running, go to a different url
```javascript
$location.path('/some-other-url');
```
  - $locationChangeSuccess is broadcasted, which closes the modal
3. async operation finishes
  - `$http.post` promise is resolved, causing `close` to be called
  - **Because cleanUpClose was called already and set everything to null, the above exception is raised**
